controllers/antreainstall_controller.go

```diff
@@ -355,3 +413,63 @@ func updateNetworkStatus(c client.Client, clusterConfig *configv1.Network, defau
 	log.Info("Successfully updated Network Status")
 	return nil
 }
+
+func updateStatusManagerAndSharedInfo(r *AntreaInstallReconciler, objs []*uns.Unstructured, clusterConfig *configv1.Network) error {
+	var daemonSets, deployments []types.NamespacedName
+	var relatedObjects []configv1.ObjectReference
+	var daemonSetObject, deploymentObject *uns.Unstructured
+	for _, obj := range objs {
+		if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "DaemonSet" {
+			daemonSets = append(daemonSets, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
+			daemonSetObject = obj
+		} else if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "Deployment" {
+			deployments = append(deployments, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
+			deploymentObject = obj
+		}
+		restMapping, err := r.Mapper.RESTMapping(obj.GroupVersionKind().GroupKind())
+		if err != nil {
+			log.Error(err, "failed to get REST mapping for storing related object")
+			continue
+		}
+		relatedObjects = append(relatedObjects, configv1.ObjectReference{
+			Group:     obj.GetObjectKind().GroupVersionKind().Group,
+			Resource:  restMapping.Resource.Resource,
+			Name:      obj.GetName(),
+			Namespace: obj.GetNamespace(),
+		})
+		if clusterConfig != nil {
+			if err := controllerutil.SetControllerReference(clusterConfig, obj, r.Scheme); err != nil {
+				log.Error(err, "failed to set owner reference", "resource", obj.GetName())
+				r.Status.SetDegraded(statusmanager.OperatorConfig, "ApplyObjectsError", fmt.Sprintf("Failed to set owner reference: %v", err))
+				return err
+			}
+		}
+	}
+	if daemonSetObject == nil || deploymentObject == nil {
+		var missedResources []string
+		if daemonSetObject == nil {
+			missedResources = append(missedResources, fmt.Sprintf("DaemonSet: %s", operatortypes.AntreaAgentDaemonSetName))
+		}
+		if deploymentObject == nil {
+			missedResources = append(missedResources, fmt.Sprintf("Deployment: %s", operatortypes.AntreaControllerDeploymentName))
+		}
+		err := fmt.Errorf("configuration of resources %v is missing", missedResources)
+		log.Error(nil, err.Error())
+		r.Status.SetDegraded(statusmanager.OperatorConfig, "ApplyObjectsError", err.Error())
+		return err
+	}
+	r.Status.SetDaemonSets(daemonSets)
+	r.Status.SetDeployments(deployments)
+	r.Status.SetRelatedObjects(relatedObjects)
+	r.SharedInfo.AntreaAgentDaemonSetSpec = daemonSetObject.DeepCopy()
+	r.SharedInfo.AntreaControllerDeploymentSpec = deploymentObject.DeepCopy()
+	return nil
+}
+
+func (a *AdaptorK8s) UpdateStatusManagerAndSharedInfo(r *AntreaInstallReconciler, objs []*uns.Unstructured, clusterConfig *configv1.Network) error {
+	return updateStatusManagerAndSharedInfo(r, objs, clusterConfig)
+}
+
+func (a *AdaptorOc) UpdateStatusManagerAndSharedInfo(r *AntreaInstallReconciler, objs []*uns.Unstructured, clusterConfig *configv1.Network) error {
+	return updateStatusManagerAndSharedInfo(r, objs, clusterConfig)
+}
```

Comment:
> It seems that these two functions are similar? Can we directly invoke `updateStatusManagerAndSharedInfo`? And the same question for `FillConfigs`, `ValidateConfig`, and `GenerateRenderData`

Good point, and this really bothered me. Taking `fillConfig` for example, I extracted code to the single function `fillConfig` to avoid code duplication. And for the case of `openshift` with network crd, we can invoke the function with `fillConfig(clusterConfig *configv1.Network, operConfig *operatorv1.AntreaInstall)`, and no problem. But for the case of `kubernetes` without network crd, the function is invoked with `fillConfig(nil, operConfig *operatorv1.AntreaInstall)`, and I think the argument `nil` may lead to confusion. Any good idea about this?